mlir/lib/Bindings/Python/IRCore.cpp

```diff
@@ -2789,6 +2786,91 @@ class PyOpAttributeMap {
   PyOperationRef operation;
 };
 
+constexpr size_t kMaxFrames = 512;
+
+MlirLocation tracebackToLocation(MlirContext ctx) {
+  size_t framesLimit =
+      PyGlobals::get().getTracebackLoc().locTracebackFramesLimit();
+  // Use a thread_local here to avoid requiring a large amount of space.
+  thread_local std::array<MlirLocation, kMaxFrames> frames;
```

Comment:
Is the below code safe in case `locTracebackFramesLimit` is set to a number > `kMaxFrames`?

Reply:
yea definitely not - std::array is statically allocated right? good catch i'll some guardrails.

Reply:
done